### PR TITLE
Change visibility modifier of DeleteCommand & SelectCommand targetIndex #527

### LINKED ISSUE #527: Change visibility modifier of DeleteCommand & SelectCommand #targetIndex
Both `DeleteCommand#targetIndex` & `SelectCommand#targetIndex` visibility modifier is `public`.

It breaks the encapsulation of `DeleteCommand` & `SelectCommand` by exposing the variable that should not be exposed to other classes.

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Fixes #527

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@Zhiyuan-Amos submitted v1 for review.

* [v1 1/2] [Implement \#equals\(Object\) for DeleteCommand and SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/0409f1d0f28e3b3fbd8f8a1021be0d47bc308922)
* [v1 2/2] [Update visibility modifier of DeleteCommand & SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/1fb71a31be9e494129816f74c7fe27def0aa5d0a)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/8cfab29d930e3fa60c7411a7eb6992373cff0686...1fb71a31be9e494129816f74c7fe27def0aa5d0a))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/528/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v2

@Zhiyuan-Amos submitted v2 for review.

* [v2 1/2] [Implement equals\(Object\) for DeleteCommand and SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/f534d07ab34e1f5ab213dacd3110a2eda8b8eae1)
* [v2 2/2] [Update visibility modifier of DeleteCommand & SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/a60232cbf84fe7cd34ed524fa3d6c91957a4b5dc)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/2c9814756eafc5121e1a13e4ce4d5b8a9b28b7bf...a60232cbf84fe7cd34ed524fa3d6c91957a4b5dc)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/e97d99d81dfe022fce5e4d0f6dbe043dea3ce096~1...e97d99d81dfe022fce5e4d0f6dbe043dea3ce096))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/528/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by Zhiyuan-Amos** [comment]
did a rebase and made the changes that yamgent requested.

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 1]
```diff
@@ -76,6 +77,30 @@ public void execute_invalidIndexFilteredList_throwsCommandException() throws Exc
...
+
+        // null -> returns false
+        assertFalse(deleteFirstCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(deleteFirstCommand.equals(deleteSecondCommand));
+    }
+
+
+
```
Comment (limmlingg) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java]: So many blank lines

File: src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
     @Test
     public void parse_validArgs_returnsDeleteCommand() throws Exception {
         DeleteCommand command = parser.parse("1");
-        assertEquals(INDEX_FIRST_PERSON, command.targetIndex);
+        assertEquals(new DeleteCommand(INDEX_FIRST_PERSON), command);
```
Comment (limmlingg) [inline src/test/java/seedu/address/logic/parser/DeleteCommandParserTest.java]: Acknowledge this change in the commit message.

**Comment by CanIHasReview** [comment]
# v3

@Zhiyuan-Amos submitted v3 for review.

* [v3 1/2] [Implement equals\(Object\) for DeleteCommand and SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/215f7f5b71324d64e0d4a44b0942001abd51e6eb)
* [v3 2/2] [Update visibility modifier of DeleteCommand & SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/f2945f71753ac7aca4ab0a77ac7b99e743f8e514)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/2c9814756eafc5121e1a13e4ce4d5b8a9b28b7bf...f2945f71753ac7aca4ab0a77ac7b99e743f8e514)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/0728e3863439484cd6742bae553902a04aa251a3~1...0728e3863439484cd6742bae553902a04aa251a3))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/528/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v4

@Zhiyuan-Amos submitted v4 for review.

* [v4 1/2] [Implement equals\(Object\) for DeleteCommand and SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/215f7f5b71324d64e0d4a44b0942001abd51e6eb)
* [v4 2/2] [Update visibility modifier of DeleteCommand & SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/1abfdd9ba72c8a67387db728fe319d9ca0077120)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/2c9814756eafc5121e1a13e4ce4d5b8a9b28b7bf...1abfdd9ba72c8a67387db728fe319d9ca0077120)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/fbd302334336609e6d3a2ebe2efc36a764c77a11~1...fbd302334336609e6d3a2ebe2efc36a764c77a11))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/528/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v5

@Zhiyuan-Amos submitted v5 for review.

* [v5 1/2] [Implement equals\(Object\) for DeleteCommand & SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/16e31fc560bc8e444312af09a5eae209cc813d6a)
* [v5 2/2] [Update visibility modifier of DeleteCommand & SelectCommand](https://github.com/se-edu/addressbook-level4/pull/528/commits/951ca32ca14f47e139a7167c2ac152f1751f6fe9)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/2c9814756eafc5121e1a13e4ce4d5b8a9b28b7bf...951ca32ca14f47e139a7167c2ac152f1751f6fe9)) ([:chart_with_upwards_trend: Interdiff between v4 and v5](https://github.com/se-edu/addressbook-level4/compare/805394f702518a0d18616f86e7b419b20ecc37a5~1...805394f702518a0d18616f86e7b419b20ecc37a5))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/528/5/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Pull request merged**